packages/store/src/internal/lifecycle-state-manager.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+
+import { filter, tap, mergeMap } from 'rxjs/operators';
+
+import { StateFactory } from './state-factory';
+import { Bootstrapper } from './bootstrapper';
+import { InternalStateOperations } from './state-operations';
+import { MappedStore, StatesAndDefaults } from './internals';
+import { LifecycleHooks, NgxsLifeCycle } from '../symbols';
+
+@Injectable()
+export class LifecycleStateManager {
+  constructor(
+    private internalStateOperations: InternalStateOperations,
+    private factory: StateFactory,
```

Comment:
```suggestion
    private stateContextFactory: StateContextFactory,
```
You should rather inject and use `StateContextFactory` here. This is the correct factory to use to create a StateContext.

Reply:
@markwhitfeld Oh, nice catch! :wink: